postem/postem-app/src/main/java/org/sakaiproject/postem/helpers/CSV.java

```diff
@@ -0,0 +1,268 @@
+import java.util.Iterator;
+import java.util.List;
+import java.util.zip.DataFormatException;
+
+public class CSV {
+
+    private List contents;
+    private List headers;
+    private List students;
+    private String csv;
+    private static boolean truncatingWarningDisplayed;
+    public static final char COMMA_DELIM = ',';
+    public static final char TAB_DELIM = '\t';
+    public static final int MAX_COL_LENGTH = 2000;
+
+    public CSV(String csv, boolean withHeader) throws DataFormatException {
+        truncatingWarningDisplayed = false;
+        contents = retrieveContents(csv, COMMA_DELIM);
+        headers = retrieveHeaders(contents);
+        students = retrieveStudents(contents, withHeader);
+        this.csv = csv;
+    }
+
+    public CSV(String csv, boolean withHeader, char delimiter) throws DataFormatException {
+        truncatingWarningDisplayed = false;
+        contents = retrieveContents(csv, delimiter);
+        headers = retrieveHeaders(contents);
+        students = retrieveStudents(contents, withHeader);
+        this.csv = csv;
+    }
+
+    public CSV(List contents, boolean withHeader) {
+        truncatingWarningDisplayed = false;
+        headers = retrieveHeaders(contents);
+        students = retrieveStudents(contents, withHeader);
+        csv = createFromContents(contents);
+        this.contents = contents;
+   }
+
+    public List getHeaders() {
+        return headers;
+    }
+
+    public List getStudents() {
+        return students;
+    }
+
+    public List getStudentUsernames() {
+        List usernames = new ArrayList();
+
+        Iterator studentIter = students.iterator();
+        while (studentIter.hasNext()) {
+            List s1 = (List) studentIter.next();
+            String usr = ((String) s1.get(0)).trim();
+            usernames.add(usr);
+            }
+
+        return usernames;
+    }
+
+    public String getCsv() {
+        return csv;
+    }
+
+    public static String createFromContents(List rows) {
+        StringBuilder csv = new StringBuilder();
+
+        Iterator riter = rows.iterator();
+        while (riter.hasNext()) {
+            Iterator citer = ((List) riter.next()).iterator();
+            while (citer.hasNext()) {
+                String current = (String) citer.next();
+                if (current == null) {
+                    current = "";
+                }
+                current = current.replaceAll("\"", "\"\"");
+                csv.append("\"");
+                csv.append(current);
+                csv.append("\"");
+
+                if (citer.hasNext()) {
+                    csv.append(",");
+                }
+            }
+            if (riter.hasNext()) {
+                csv.append("\r\n");
+            }
+        }
+        return csv.toString();
+    }
+
+    public static List retrieveContents(String csv, char delimiter) throws DataFormatException {
+
+        List all = new ArrayList();
+        List current = new ArrayList();
```

Comment:
Are these lists of String? You might as well type them for clarity.

Reply:
I've been doing a deep review of the collections and types in the classes and have thoroughly modified many of the lists. However, I am partly using legagy code at some points and the changes have been made as far as I can. Thanks a lot.